cmd/exec/exec_docker.go

```diff
@@ -16,6 +16,15 @@ import (
 	"github.com/iximiuz/cdebug/pkg/uuid"
 )
 
+// imageExistsLocally checks if an image exists in the host
+func imageExistsLocally(ctx context.Context, client *docker.Client, image string) bool {
+	_, _, err := client.ImageInspectWithRaw(ctx, image)
```

Comment:
There might be a catch here. We should probably check that the found image is of the expected platform. Currently, the _expected platform_ is defined as `opts.platform || target.Platform || defaultPlatform`.

Reply:
I think `platform` might be useful in the containerd impl but docker's client.ImageInspectWithRaw didn't need it

Reply:
I'm afraid we still should be checking the `platform` even with Docker. Here is a small experiment on and amd64 machine (using the code from this PR):

```
# Pre-pull the debugger image but intentionally using the "wrong" platform.
$ docker pull --platform linux/arm64 busybox:musl

# Start a target container
$ docker run --platform linux/amd64 nginx

# cdebug exec won't pull the busybox image. However,
# the cdebug's binaries will be of another architecture
# than the target image. And it may or may not be a problem.
$ ./cdebug exec -it 3ffb9dda355a
> ps auxf
PID   USER     TIME  COMMAND
    1 root      0:00 nginx: master process nginx -g daemon off;
   29 nginx     0:00 nginx: worker process
   30 nginx     0:00 nginx: worker process
   31 nginx     0:00 nginx: worker process
   32 nginx     0:00 nginx: worker process
   33 root      0:00 {sh} /usr/libexec/qemu-binfmt/aarch64-binfmt-P /bin/sh sh /.cdebug-entrypoint.sh
   47 root      0:00 sh
   50 root      0:00 ps auxf
   ```
   
Notice the `/usr/libexec/qemu-binfmt/aarch64-binfmt-P` part in the above output.
   
Would be great to have the debugger sidecar of the same architecture as the target container to avoid any potentially weird behaviours.

Reply:
Thanks for the feedback and for letting me know about lima, I made some changes and updated the description with the tests that I run considering platform